### PR TITLE
Unit test for classes

### PR DESCRIPTION
**Pull request by cilfonegabriel**
# Dear Code Reviewers.
## Could I ask you for your kindly support to review my Project? 🙏

**⚠️ NOTE: There are commits only by myself since I am working solo for this week. You can assess this on [this comment](https://github.com/cilfonegabriel/OOP-school-library/pull/8#issuecomment-1480324878)**

### Project requirements
- Add unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - `Nameable`
  - `CapitalizeDecorator`
  - `Decorator`

### Thank´s Gabriel!

### CODE REVIEW

**Review by Whoistolu — COMMENTED** [review]
Hi @cilfonegabriel 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a pair-programming project, that should have contributions from you and your partner(s) but you are the only developer who has contributed to this project. As much as I understand that your partner dropped off, it is required to get approval to work alone from the student success team and attach a screenshot of the response to your PR description.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by rmrmrmr** [comment]
## Hi @Whoistolu and future code reviewer!
I can validate that although this project is marked as pair-programming @cilfonegabriel is working solo. I was assigned as his coding partner at the beginning of the week, but it was only temporary since I moved on to the next one. We had no other partners so he can't include commits of any other non-existing micronauts! 😝 
 Here are some links to the [repo I've been working on](https://github.com/Gdiazdiaz/Capstone-catalogue-of-my-things) and to some[ pull requests I've opened there](https://github.com/Gdiazdiaz/Capstone-catalogue-of-my-things/pull/36) so you can assess that indeed I'm no longer his coding partner.

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        expect(person.can_use_services?).to be true
+      end
+    end
+  end
+
+  describe '#to_json' do
+    it 'returns a JSON representation of the person' do
+      expect(person.to_json).to eq({
+        JSON.create_id => 'Person',
+        'id' => '1234-5678',
```
Comment (BenMukebo) [inline spec/person_spec.rb]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `of_age?, correct_name and add_rental` class person method has (at least) an 'it' block 👇🏻 check your person.rb file

```ruby
 def of_age?
    @age > 18
  end

  def correct_name
    @name
  end

  def add_rental(book, date = Date.today)
    @rentals.push(Rental.new(date, self, book))
  end
```

File: spec/decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require_relative './../modules/decorator'
+require_relative './../modules/nameable'
+
+describe '#new' do
+  decorator = Decorator.new('label')
+  it 'takes a new decorator object' do
+    expect(decorator).to be_instance_of(Decorator)
+  end
+
+  it 'returns a nameable object' do
```
Comment (BenMukebo) [inline spec/decorator_spec.rb]: - [ ] Kindly make sure to add unit tests for all the methods of the: `Decorator class` ✍🏻 Cureent you're missing the `correct_name` method tests. 👇🏻  


```ruby
  def correct_name
    @nameable.correct_name
  end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @cilfonegabriel,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- All test files is placed in a different directory. ✔️  
- Gitflow was correctly used.✔️ 
- All tests are passing. 🥇 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL ] Use descriptive commit messages.  The commit message should be written in the imperative tense and capitalized. [Read more about how to write descriptive commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8). Example:
  - :x: Wrong: added Descriptive README
  - :heavy_check_mark: Right: Add descriptive README

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      it 'returns false' do
+        expect(person.can_use_services?).to be false
+      end
+    end
+  end
+
+  describe '#correct_name' do
+    it 'returns the name attribute' do
+      expect(person.correct_name).to eq('Unknown')
+    end
```
Comment (Reem-lab) [inline spec/person_spec.rb]: - [ ] Great on working on the changes that requested from the previous reviewer but I think you miss understand what he asked you, in Person you missed to add tests to the methods that he mentioned `of_age?, correct_name and add_rental` so great you added for `correct_name` but you still missed to test the another two methods and he said that (at least) an 'it' block 👇🏻  but it block for each method so at least one test for each method not at least one method 🙏🏻✅💯🤩 so kindly add tests for the another methods  `of_age? and add_rental` 💪🏻

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi Team, 👋🏻👋🏻

Great work on making the changes requested by a previous reviewer 👏💯🔥✨🤩🤩 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




 ## Highlights
- No Linters error ✨🤩
- Great working on the tests ✨💯
- Followed Gitflow 🤩🔥🔥

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @cilfonegabriel , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**